doc/services/pm/evtc.rst

```diff
@@ -0,0 +1,189 @@
+
+EVT latency
+***********
+
+The EVT latency is the time from a hardware event originating until
+it reaches its destination.
+
+Examples of EVT latencies:
+
+* From GPIO state changed to interrupt being triggered.
+* From interrupt triggered to interrupt being serviced.
+
+PM EVTC
+*******
+
+PM EVTCs manage EVT latencies.
+
+Examples of PM EVTCs:
+
+* Nordic power device (:dtcompatible:`nordic,nrf-power`)
+* Nordic GPIOTE device (:dtcompatible:`nordic,nrf-gpiote`)
+* Nordic DPPIC device (:dtcompatible:`nordic,nrf-dppic`)
+
+Predictable EVT latency management
+**********************************
+
+Purpose
+-------
+
+To minimize latency just in time for an EVT to originate, and
+maximize latency right after the EVT has been serviced.
+
+Usage
+-----
+
+1. User configures EVT originator to originate EVT at predictable
+   time.
+2. User schedules PM EVT with PM EVTC using
+   :c:func:`pm_evtc_schedule_evt()`, specifying:
+   * A time shortly before the EVT is predicted to originate.
+   * The maximum acceptable latency of the EVT.
+3. PM EVTC minimizes EVT latency just in time for it to be in
+   effect before the event is predicted to originate.
+4. EVT originates
+5. EVT is handled by the user
+6. User releases scheduled PM EVT using
+   :c:func:`pm_evtc_release_evt()`
+7. PM EVTC maximizes EVT latency.
+
+The time spent with minimized latency, and thus higher energy
+consumption, is minimized to just before the event to just after
+the event has been handled, as notified by the user.
+
+A PM EVT can be additionally be rescheduled using
+:c:func:`pm_evtc_reschedule_evt()` which implicitly and efficiently
+releases the PM EVT followed by scheduling it.
+
+Unpredictable EVT latency management
+************************************
+
+Purpose
+-------
+
+To ensure a maximum EVT latency while maximizing EVT latency.
+
+Usage
+-----
+
+1. User configures EVT originator to originate EVTs
+2. User requests maximum EVT latency with PM EVTC using
+   :c:func:`pm_evtc_request_evt()`.
+3. PM EVTC minimizes EVT latency immediately to be in effect as "soon
+   as possible".
+
+While the EVT latency is in effect, it is not neccesarily constant,
+but will not exceed the requested maximum.
+
+4. EVT originates.
+5. EVT is handled by the user.
+
+The above steps are repeated until the user no longer requires a
+maximum EVT latency.
+
+6. User releases EVT using :c:func:`pm_evtc_release_evt()`.
+7. PM EVTC maximizes EVT latency.
+
+PM EVTC implementation
+**********************
+
+Devicetree
+----------
+
+Event controllers which support power management must include the base
+binding ``evtc.yaml``:
+
+.. code-block:: yaml
+
+   include:
+     - "evtc.yaml"
+
+which includes the required properties ``evtc-request-latency-ns`` and
+``evtc-latencies-ns`` which must be defined in the devicetree:
+
+.. code-block:: devicetree
+
+   evtc0: evtc0 {
+           /* Latency for request to take effect in nanoseconds */
+           evtc-request-latency-ns = <3500>;
```

Comment:
These properties are not clear to me. Is `evtc-request-latency-ns` hw property  or it depends in the sw implementation ? Or it depends on what the driver will do when the latency is requested ?

Reply:
It is a hardware property, how long does it take from requesting some latency for it to be in effect. Its effectively the time it takes to switch power modes.